src/client/datascience/kernel-launcher/kernelLauncher.ts

```diff
@@ -151,10 +153,18 @@ export class KernelLauncher implements IKernelLauncher {
         this.portChain = this.getConnectionPorts();
         return this.portChain;
     }
+    static async findNextFreePort(port: number): Promise<number[]> {
```

Comment:
Tests were failing 90% of the time for me due to addresses being in use.
This was a simple way to get around it.